packages/jupyterlab-manager/src/manager.ts

```diff
@@ -614,7 +616,12 @@ export class WidgetManager extends LabWidgetManager {
   ): Promise<void> {
     try {
       if (loadKernel) {
-        await this._loadFromKernel();
+        try {
+          this._kernelRestoreInProgress = true;
+          await this._loadFromKernel();
+        } finally {
+          this._kernelRestoreInProgress = false;
+        }
```

Comment:
Would it make more sense to put the setting/unsetting of this flag inside the method?

Reply:
Can you be more explicit about what you have in mind?